### ISSUE TITLE
println doesn't appear in logcat output

### ISSUE THREAD

**Issue by jdm** [servo/servo#21637]
This makes it harder to interact with code that is already built around println, like various `-Z` options. Is there some way to redirect stdout to logcat, @paulrouget? Or perhaps @fabricedesre knows?

**Comment by fabricedesre** [comment]
The android_glue crate used to redirect stdout to logcat. I don't know if things have changed there..

**Comment by jdm** [comment]
Aha. Since we stopped using that crate, we no longer get that. We'll want to do something like https://github.com/tomaka/android-rs-glue/blob/0e80cfbcee362f207389dc2660525c3d74987f70/cargo-apk/injected-glue/lib.rs#L240-L303 I suppose.

**Comment by SimonSapin** [comment]
This was a know regression of https://github.com/servo/servo/pull/20912. See also https://github.com/servo/servo/issues/19999.

In the meantime, the output of macros from the `log` crate should show up in logcat.

**Comment by SimonSapin** [comment]
Yes, it wont catch everything going to stdio file descriptors like android-rs-glue did but I think it should catch Rust `println!`, assertions, panics, etc.

**Comment by jdm** [comment]
set_print appears to only affect the thread that calls it, so that doesn't look like a feasible solution.

**Comment by paulrouget** [comment]
So I guess the best way is to reddirect stdout. https://github.com/tomaka/android-rs-glue/blob/0e80cfbcee362f207389dc2660525c3d74987f70/cargo-apk/injected-glue/lib.rs#L250
